test/sync/src/androidTest/kotlin/io/realm/test/mongodb/shared/SyncSessionTests.kt

```diff
@@ -110,6 +122,199 @@ class SyncSessionTests {
+            // being uploaded and it not being immediately ready for download
+            // on another Realm. In order to reduce the flakyness, we are
+            // re-evaluating the assertion multiple times.
+            for (i in 4 downTo 0) {
+                assertTrue(realm2.syncSession.downloadAllServerChanges())
+                val size = realm2.query<ParentPk>().count().find()
+                when (size == 10L) {
+                    true -> break // Test succeeded
+                    false -> {
+                        if (i == 0) {
+                            throw kotlin.AssertionError("Realm failed to receive download data: $size")
+                        }
+                        if (size != 0L) {
+                            throw AssertionError("Unexpected size: $size")
+                        }
+                        delay(100)
+                    }
+                }
+            }
+        } finally {
+            realm1.close()
+            realm2.close()
+        }
+    }
+
+    @Test
+    fun uploadAndDownload_throwsInsideSyncErrorHandler() = runBlocking {
+        val channel = Channel<SyncSession>(1)
+        var wrongSchemaRealm: Realm? = null
+        val job = async {
+
+            // Create server side Realm with one schema
+            var config = SyncConfiguration.Builder(user, DEFAULT_PARTITION_VALUE, schema = setOf(ParentPk::class, ChildPk::class))
+                .directory(tmpDir)
+                .build()
+            val realm = Realm.open(config)
+            realm.syncSession.uploadAllLocalChanges()
+            realm.close()
+
+            // Create same Realm with another schema, which will cause a Client Reset.
+            config = SyncConfiguration.Builder(user, DEFAULT_PARTITION_VALUE, schema = setOf(ParentPk::class, io.realm.entities.sync.bogus.ChildPk::class))
+                .directory(tmpDir)
+                .name("new_realm.realm")
+                .errorHandler(object : SyncSession.ErrorHandler {
+                    override fun onError(session: SyncSession, error: SyncException) {
+                        channel.trySend(session)
+                    }
+                })
+                .build()
+            wrongSchemaRealm = Realm.open(config)
+        }
+        val session = channel.receive()
+        try {
+            assertFailsWith<IllegalStateException> {
+                session.uploadAllLocalChanges()
+            }.also {
+                assertTrue(it.message!!.contains("Uploading and downloading changes is not allowed"))
+            }
+            assertFailsWith<IllegalStateException> {
+                session.downloadAllServerChanges()
+            }.also {
+                assertTrue(it.message!!.contains("Uploading and downloading changes is not allowed"))
+            }
+        } finally {
+            wrongSchemaRealm?.close()
+            job.cancel()
+            channel.close()
+        }
+        Unit
```

Comment:
```suggestion
```

Reply:
We need to return Unit here as test methods are not allowed to return anything and with this pattern `fun method() = runBlocking { }`, it will return whatever is returned from runBlocking.